### PR TITLE
Feature/pythonscripts

### PR DESCRIPTION
**Pull request by nyuuyn**
#### Short Description
Adds feature to use python scripts instead of bash scripts for configuring debian based components

#### Changes
Enables the following:
  * Add a single zip file with the following structure (ArtifactType: {http://opentosca.org/artifacttypes}PythonArchiveArtifact_w1-wip1): 
    - Single *.py file in root
    - Single requirements.txt file in root
    - Anything else can be in subfolders
  * Add a single python 3 script file as a file (ArtifactType:  {http://opentosca.org/artifacttypes}PythonScriptArtifact_w1-wip1)

### CODE REVIEW

**Comment by nyuuyn** [comment]
Things that still have to be done:

Installing python before executing the script:
However, there is a slight problem with, e.g., docker containers. The invocation plugin expects an installPackage method on the interface which is not provided at this time. Maybe installing it via the runScript method? (apt-get update && apt-get install python)

Output handling (maybe only in Docker Container):
The results from the runScript method are returning the script log as a single string without new lines and so on, therefore, the current output parsing doesn't work.

Example expected Output:
Output1=asd
Output2=qwe

Returned output:
Output1=asdOutput2=qwe

**Comment by nyuuyn** [comment]
For testing, I created a small dummy topology, which "should" install an apache webserver on an ubuntu running as a docker container.

https://drive.google.com/file/d/1hRhwDevRnkRCJGD5CuwrtiZzOyrKDoDq/view?usp=sharing

**Comment by nyuuyn** [comment]
Added final version for python scripts as IAs

We should be able to use bare python scripts or packaged as zips with requirements.txt and more files

Updated Test CSAR:
https://drive.google.com/file/d/1hRhwDevRnkRCJGD5CuwrtiZzOyrKDoDq/view?usp=sharing

**Comment by lharzenetter** [comment]
Can you please create a PR with the Apache implementation in <https://github.com/OpenTOSCA/tosca-definitions-common>?

File: org.opentosca.bus/org.opentosca.bus.management.deployment.plugin.script/src/main/java/org/opentosca/bus/management/deployment/plugin/script/ManagementBusDeploymentPluginScript.java [hunk 1]
```diff
@@ -25,7 +25,7 @@
...
+    static final private String TYPES = "{http://opentosca.org/artifacttypes}PythonArchiveArtifact_w1-wip1, {http://opentosca.org/artifacttypes}PythonScriptArtifact_w1-wip1, {http://opentosca.org/artifacttypes}Chef, {http://opentosca.org/artifacttypes}Ansible, {http://www.example.com/clartigr/tosca}clartigr, {http://docs.oasis-open.org/tosca/ns/2011/12/ToscaBaseTypes}Chef, {http://docs.oasis-open.org/tosca/ns/2011/12/ToscaBaseTypes}Puppet, {http://docs.oasis-open.org/tosca/ns/2011/12/ToscaBaseTypes}Juju, {http://docs.oasis-open.org/tosca/ns/2011/12/ToscaBaseTypes}CloudFoundry, {http://docs.oasis-open.org/tosca/ns/2011/12/ToscaBaseTypes}UnixShell, {http://docs.oasis-open.org/tosca/ns/2011/12/ToscaBaseTypes}ScriptArtifact";
```
Comment (lharzenetter) [inline org.opentosca.bus/org.opentosca.bus.management.deployment.plugin.script/src/main/java/org/opentosca/bus/management/deployment/plugin/script/ManagementBusDeploymentPluginScript.java:28]: Maybe list the available types in separate lines for better readability.
Also, remove the version suffixes. However, I would suggest to create an abstract python Artifact Type and create subtypes of it with versioning.

File: org.opentosca.bus/org.opentosca.bus.management.invocation.plugin.script/src/main/resources/artifacttypes/PythonArchive.xml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+	namespace="http://opentosca.org/artifacttypes">
+	<packages>
+        <package>unzip</package>
+        <package>zip</package>
+        <package>python3</package>
+        <package>python3-pip</package>
+	</packages>
+	<commands>
+		<command>chmod +x {TARGET_FILE_PATH}</command>
+		<!--<command>CAN_SUDO=$(sudo -n uptime 2&gt;&amp;1|grep "load"|wc -l); if [ ${CAN_SUDO} -gt 0 ]; then sudo -E {DA_NAME_PATH_MAP}{INPUT_PARAMETER} python {TARGET_FILE_PATH}; else {DA_NAME_PATH_MAP}{INPUT_PARAMETER} python {TARGET_FILE_PATH}; fi</command>-->
```
Comment (lharzenetter) [inline org.opentosca.bus/org.opentosca.bus.management.invocation.plugin.script/src/main/resources/artifacttypes/PythonArchive.xml:12]: What is with the can sudo prequel?

File: org.opentosca.bus/org.opentosca.bus.management.invocation.plugin.script/src/main/resources/artifacttypes/PythonScript.xml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<artifacttype name="PythonScriptArtifact_w1-wip1"
+	namespace="http://opentosca.org/artifacttypes">
+	<packages>
+        <package>python3</package>
+        <package>python3-pip</package>
+	</packages>
+	<commands>
+		<command>chmod +x {TARGET_FILE_PATH}</command>
+		<!--<command>CAN_SUDO=$(sudo -n uptime 2&gt;&amp;1|grep "load"|wc -l); if [ ${CAN_SUDO} -gt 0 ]; then sudo -E {DA_NAME_PATH_MAP}{INPUT_PARAMETER} python {TARGET_FILE_PATH}; else {DA_NAME_PATH_MAP}{INPUT_PARAMETER} python {TARGET_FILE_PATH}; fi</command>-->
```
Comment (lharzenetter) [inline org.opentosca.bus/org.opentosca.bus.management.invocation.plugin.script/src/main/resources/artifacttypes/PythonScript.xml:10]: See above

**Review by lharzenetter — CHANGES_REQUESTED** [review]

File: org.opentosca.bus/org.opentosca.bus.management.deployment.plugin.script/src/main/java/org/opentosca/bus/management/deployment/plugin/script/ManagementBusDeploymentPluginScript.java (hunk 1 above)
Comment (nyuuyn) [reply org.opentosca.bus/org.opentosca.bus.management.deployment.plugin.script/src/main/java/org/opentosca/bus/management/deployment/plugin/script/ManagementBusDeploymentPluginScript.java:28]: Does the hierarchy stuff even work ?

File: org.opentosca.bus/org.opentosca.bus.management.invocation.plugin.script/src/main/resources/artifacttypes/PythonScript.xml (hunk 3 above)
Comment (nyuuyn) [reply org.opentosca.bus/org.opentosca.bus.management.invocation.plugin.script/src/main/resources/artifacttypes/PythonScript.xml]: fixec

File: org.opentosca.bus/org.opentosca.bus.management.invocation.plugin.script/src/main/resources/artifacttypes/PythonArchive.xml (hunk 2 above)
Comment (nyuuyn) [reply org.opentosca.bus/org.opentosca.bus.management.invocation.plugin.script/src/main/resources/artifacttypes/PythonArchive.xml]: fixed

File: org.opentosca.bus/org.opentosca.bus.management.invocation.plugin.script/src/main/java/org/opentosca/bus/management/invocation/plugin/script/ManagementBusInvocationPluginScript.java [hunk 4]
```diff
@@ -395,13 +395,11 @@ private void installPackages(final TArtifactType artifactType, final Map<String,
             LOG.debug("ArtifactType: {} needs no packages to install.", requiredPackages, artifactType);
             return;
         }
+        
         final String requiredPackagesString = String.join(" ", requiredPackages);
-        final Map<String, String> inputParamsMap = new HashMap<>();
-        inputParamsMap.put(Interfaces.OPENTOSCA_DECLARATIVE_INTERFACE_OPERATINGSYSTEM_PARAMETER_PACKAGENAMES, requiredPackagesString);
+        final String commandsString = "apt update && export DEBIAN_FRONTEND=noninteractive && apt install -y -q " + requiredPackagesString;
```
Comment (zimmerml) [inline org.opentosca.bus/org.opentosca.bus.management.invocation.plugin.script/src/main/java/org/opentosca/bus/management/invocation/plugin/script/ManagementBusInvocationPluginScript.java:400]: Isn't that an extreme assumption?

File: org.opentosca.bus/org.opentosca.bus.management.deployment.plugin.script/src/main/java/org/opentosca/bus/management/deployment/plugin/script/ManagementBusDeploymentPluginScript.java (hunk 1 above)
Comment (lharzenetter) [reply org.opentosca.bus/org.opentosca.bus.management.deployment.plugin.script/src/main/java/org/opentosca/bus/management/deployment/plugin/script/ManagementBusDeploymentPluginScript.java]: Depends on where the type is checked. However, I'd still remove the `w1-wip` in this case.

**Review by zimmerml — APPROVED** [review]

File: org.opentosca.bus/org.opentosca.bus.management.deployment.plugin.script/src/main/java/org/opentosca/bus/management/deployment/plugin/script/ManagementBusDeploymentPluginScript.java (hunk 1 above)
Comment (nyuuyn) [reply org.opentosca.bus/org.opentosca.bus.management.deployment.plugin.script/src/main/java/org/opentosca/bus/management/deployment/plugin/script/ManagementBusDeploymentPluginScript.java]: fixed

File: org.opentosca.bus/org.opentosca.bus.management.invocation.plugin.script/src/main/java/org/opentosca/bus/management/invocation/plugin/script/ManagementBusInvocationPluginScript.java (hunk 4 above)
Comment (nyuuyn) [reply org.opentosca.bus/org.opentosca.bus.management.invocation.plugin.script/src/main/java/org/opentosca/bus/management/invocation/plugin/script/ManagementBusInvocationPluginScript.java:400]: yes

**Review by lharzenetter — APPROVED** [review]

**Pull request merged**